client/service/src/chain_ops.rs

```diff
@@ -37,12 +38,183 @@ use sp_core::storage::{StorageKey, well_known_keys, ChildInfo, Storage, StorageC
+	where
+		R: Read + Seek + 'static,
+		B: BlockT + MaybeSerializeDeserialize,
+	{
+	type Item = Result<SignedBlock<B>, String>;
+
+	/// Returns Poll::Ready(None) if every block have been read,
+	/// else returns Poll::Ready(Some(block)), block being a result of decoding a signed block.
+	fn poll_next(self: Pin<&mut Self>, _cx: &mut Context) -> Poll<Option<Self::Item>> {
+		match Pin::get_mut(self) {
+			BlockStream::Binary {count, read_block_count, reader} => {
+				if read_block_count < count {
+					let block_result: Result<SignedBlock::<B>, _> =  SignedBlock::<B>::decode(reader)
+						.map_err(|e| e.to_string());
+					*read_block_count += 1;
+					Poll::Ready(Some(block_result))
+				} else {
+					// `read_block_count` == `count` so we've read enough blocks.
+					Poll::Ready(None)
+				}
+			}
+			BlockStream::Json {reader, read_block_count} => {
+				match reader.next() {
+					Some(block) => {
+						*read_block_count += 1;
+						Poll::Ready(Some(block.map_err(|e| e.to_string())))
+					}
+					None => {
+						// We've reached the end of the iterator.
+						Poll::Ready(None)
+					}
+				}
+			}
+		}
+	}
+}
+
+/// Imports the SignedBlock to the queue.
+fn import_block_to_queue<TBl, TImpQu>(signed_block: SignedBlock<TBl>, queue: &mut TImpQu, force: bool)
+where
+	TBl: BlockT + MaybeSerializeDeserialize,
+	TImpQu: 'static + ImportQueue<TBl>,
+{
+	let (header, extrinsics) = signed_block.block.deconstruct();
+	let hash = header.hash();
+	// import queue handles verification and importing it into the client.
+	queue.import_blocks(BlockOrigin::File, vec![
+		IncomingBlock::<TBl> {
+			hash,
+			header: Some(header),
+			body: Some(extrinsics),
+			justification: signed_block.justification,
+			origin: None,
+			allow_missing_state: false,
+			import_existing: force,
+		}
+	]);
+}
+
+/// Returns true if we have imported every block we were supposed to import, else returns false.
+fn importing_is_done(count: Option<u64>, read_block_count: u64, imported_blocks: u64) -> bool {
+	if let Some(count) = count {
+		if imported_blocks >= count {
+			return true
+		}
+	} else {
+		if imported_blocks >= read_block_count {
+			return true
+		}
+	}
+	false
+}
+
+// Logs information regarding the current importing status.
+fn log_importing_status_updates(count: Option<u64>, read_block_count: u64, blocks_before: u64, imported_blocks: u64) {
+	// Notify every 1000 blocks to let the user know everything is running smoothly.
+	if read_block_count % 1000 == 0 {
+		info!(
+			"#{} blocks were added to the queue",
+			read_block_count
+		);
+	}
+
+	// Notify every time we import an additional 1000 blocks.
+	if imported_blocks / 1000 != blocks_before / 1000 {
+		if let Some(count) = count {
+			info!(
+				"#{} blocks were imported (#{} left)",
+				imported_blocks,
+				count - imported_blocks
+			)
+		} else {
+			info!(
+				"{} blocks were imported, other are still being processed",
+				imported_blocks,
+			)
+		}
+	}
+}
+
```

Comment:
Those functions are helper functions to make code more readable: is there any particular guideline I should follow when writing those?

Reply:
As long as they are not `pub` I think you're pretty much free to make these helpers in the way that makes the most sense to you (somebody corrects me if I'm wrong)